agreement/asyncVoteVerifier.go

```diff
@@ -58,6 +59,15 @@ type AsyncVoteVerifier struct {
 	ctxCancel       context.CancelFunc
 }
 
+// LedgerDroppedRoundError is an error for when agreement far behind ledger
+type LedgerDroppedRoundError struct {
+	Err error
+}
+
+func (e *LedgerDroppedRoundError) Error() string {
+	return e.Err.Error()
+}
+
```

Comment:
This would be  better placed in abstract.go, where the interfaces are. We should also add some more detailed documentation for the reasons this error is used ( notify the agreement that the requested round cannot, and will not be available as it's too far back ), and why ( the agreement need to make the distinction between a malicious vote and a vote that it cannot verify ).

Reply:
Also, since you've created such a nice internal error, why don't you export it via Unwrap ? it will avoid breaking the chain.